### PR TITLE
Fix wrong path separator in error message when saved model does not exist

### PR DESCRIPTION
**Pull request by spopovru**
Error message generation changed for `keras.models.load_model` (fixed bug #44739).
Path separator recived from os.path.sep, and now for Windows error message looking like
`OSError: SavedModel file does not exist at: C:\IncorrectPathToSavedModel\{saved_model.pbtxt|saved_model.pb}`
for Linux - 
`OSError: SavedModel file does not exist at: ~/IncorrectPathToSavedModel/{saved_model.pbtxt|saved_model.pb}`

### CODE REVIEW

**Comment by spopovru** [comment]
@googlebot I signed it!

**Review by bhack — APPROVED** [review]

**Review by fchollet — CHANGES_REQUESTED** [review]
Looks good, please add a quick unit test.

**Comment by bhack** [comment]
> Looks good, please add a quick unit test.

Yes I think that it is better to have a quick unit test if he have enough time but also it is really trivial and it will be quite hard to create a regression there after this change.

**Comment by spopovru** [comment]
> Looks good, please add a quick unit test.

Thank you!
I added a test for this bug, hopefully now all right.

File: tensorflow/python/saved_model/loader_test.py [hunk 1]
```diff
@@ -290,6 +290,13 @@ def test_load_saved_model_graph_with_return_elements(self, builder_cls):
     with self.assertRaisesRegex(ValueError, "not found in graph"):
       loader.load_graph(graph, ["foo_graph"], return_elements=["z:0"])
 
+  def test_parse_saved_model_exception(self, builder_cls):
+    """Test that error message for not exist model have OS-depend delimiter in path"""
+    path = _get_export_dir("not_existing_dir")
+    pattern = os.path.sep + "{"
+    with self.assertRaises(IOError) as err:
+      loader_impl.parse_saved_model(path)
+    self.assertTrue(pattern in str(err.exception))
```
Comment (bhack) [inline tensorflow/python/saved_model/loader_test.py:299]: Is this not going to match `os.path.sep` in `_get_export_dir` and so in `test.get_temp_dir()`?

Comment (spopovru) [reply tensorflow/python/saved_model/loader_test.py:299]: No, in this test we search correct delimiter between directory name and "`{`".
Error message formed by pattern "`SavedModel file does not exist at: %s%s{%s|%s}" (in current version "SavedModel file does not exist at: %s/{%s|%s}`").
For example, "`SavedModel file does not exist at: C:\IncorrectPathToSavedModel\{saved_model.pbtxt|saved_model.pb}`".
We checking combination from delimiter and "`{`", path to temporary dir can't have this combination, it's simple test for correct path separator detection.
IMHO, this change can to be accepted without new tests, but I respect rules of project and created test for regression. For current "main" branch this test is red, for updated code - green, I think it's correct.
Excuse me if I incorrect understand your comment and give not valid answer, I want to make my code better, if I can.

Comment (bhack) [reply tensorflow/python/saved_model/loader_test.py:299]: Ok `os.path.sep + "{"`

**Review by bhack — APPROVED** [review]

**Comment by spopovru** [comment]
@fchollet can you approve this pull request, please?
I was resolve issue and add simple unit test.
It's not seriously bug, off course, but this fix make project better, imho.

**Review by fchollet — APPROVED** [review]
LGTM

**Pull request merged**